components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java

```diff
@@ -273,4 +231,336 @@ private String getParentDomainFromUrl(String url) throws URISyntaxException {
+     * @throws SecretManagementException {@link SecretManagementException}
+     * @throws IOException               {@link IOException}
+     * @throws FrameworkException        {@link FrameworkException}
+     */
+    private void requestAccessToken(String tenantDomain, AccessTokenRequestHelper accessTokenRequestHelper)
+                                    throws IOException, FrameworkException {
+
+        HttpPost request = new HttpPost(ConfigProvider.getInstance().getChoreoTokenEndpoint());
+        request.setHeader(ACCEPT, TYPE_APPLICATION_JSON);
+        request.setHeader(CONTENT_TYPE, TYPE_FORM_DATA);
+
+        request.setHeader(AUTHORIZATION, BASIC + Base64.getEncoder()
+                .encodeToString((accessTokenRequestHelper.consumerKey + ":" + accessTokenRequestHelper.consumerSecret)
+                .getBytes(StandardCharsets.UTF_8)));
+
+        List<BasicNameValuePair> bodyParams = new ArrayList<>();
+        bodyParams.add(new BasicNameValuePair(GRANT_TYPE, GRANT_TYPE_CLIENT_CREDENTIALS));
+        request.setEntity(new UrlEncodedFormEntity(bodyParams));
+
+        CloseableHttpAsyncClient client = ChoreoFunctionServiceHolder.getInstance().getClientManager()
+                .getClient(tenantDomain);
+        client.execute(request, accessTokenRequestHelper);
+    }
+
+    private class AccessTokenRequestHelper implements FutureCallback<HttpResponse> {
+
+        private final Map<String, String> connectionMetaData;
+        private final AsyncReturn asyncReturn;
+        private final AuthenticationContext authenticationContext;
+        private final Map<String, Object> payloadData;
+        private final Gson gson;
+        private final AtomicInteger tokenRequestAttemptCount;
+        private String consumerKey;
+        private String consumerSecret;
+
+        public AccessTokenRequestHelper(Map<String, String> connectionMetaData,
+                                        AsyncReturn asyncReturn,
+                                        AuthenticationContext authenticationContext,
+                                        Map<String, Object> payloadData) throws SecretManagementException {
+
+            this.connectionMetaData = connectionMetaData;
+            this.asyncReturn = asyncReturn;
+            this.authenticationContext = authenticationContext;
+            this.payloadData = payloadData;
+            this.gson = new GsonBuilder().create();
+            this.tokenRequestAttemptCount = new AtomicInteger(0);
+            resolveConsumerKeySecrete();
+        }
+
+        /**
+         * The method to be called when access token request receives an HTTP response.
+         *
+         * @param httpResponse Received HTTP response.
+         */
+        @Override
+        public void completed(HttpResponse httpResponse) {
+
+            boolean isFailure = false;
+            try {
+                LOG.debug("Access token response received.");
```

Comment:
Let's check whether the debug is enabled

Reply:
See: https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/112/files#r1116190486